### PR TITLE
op_mode: T6181: A feature for checking ports

### PR DESCRIPTION
**Pull request by natali-rs1985**
## Change Summary

Added op mode command to scan specific host for open ports
```
execute port-scan host <target> [ <Options> ]
```
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [ ] Other (please describe):

## Related Task(s)

* https://vyos.dev/T6181
## Component(s) name

op_mode: ```execute port-scan ```
## How to test

op mode command:
```
vyos@vyos:~$ execute port-scan host 192.0.2.2
Starting Nmap 7.93 ( https://nmap.org ) at 2024-09-05 08:18 UTC
Nmap scan report for 192.0.2.2
Host is up (0.0018s latency).
Not shown: 65533 closed tcp ports (conn-refused)
PORT    STATE SERVICE
22/tcp  open  ssh
443/tcp open  https

Nmap done: 1 IP address (1 host up) scanned in 17.84 seconds
```
With options:
```
vyos@vyos:~$ execute port-scan host 192.0.2.2
Possible completions:
  <Enter>               Execute the current command
  ipv6                  Port scan options
  port
  skip-ping
  tcp
  udp

vyos@vyos:~$ execute port-scan host 192.0.2.2 skip-ping tcp port 22,100-105,440-450
Starting Nmap 7.93 ( https://nmap.org ) at 2024-09-05 08:20 UTC
Nmap scan report for 192.0.2.2
Host is up (0.0010s latency).

PORT    STATE  SERVICE
22/tcp  open   ssh
100/tcp closed newacct
101/tcp closed hostname
102/tcp closed iso-tsap
103/tcp closed gppitnp
104/tcp closed acr-nema
105/tcp closed csnet-ns
440/tcp closed sgcp
441/tcp closed decvms-sysmgt
442/tcp closed cvc_hostd
443/tcp open   https
444/tcp closed snpp
445/tcp closed microsoft-ds
446/tcp closed ddm-rdb
447/tcp closed ddm-dfm
448/tcp closed ddm-ssl
449/tcp closed as-servermap
450/tcp closed tserver

Nmap done: 1 IP address (1 host up) scanned in 13.04 seconds
```
## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [x] I have linked this PR to one or more Phabricator Task(s)
- [ ] I have run the components [**SMOKETESTS**](https://github.com/vyos/vyos-1x/tree/current/smoketest/scripts/cli) if applicable
- [x] My commit headlines contain a valid Task id
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: op-mode-definitions/execute-port-scan.xml.in [hunk 1]
```diff
@@ -0,0 +1,37 @@
+<?xml version="1.0"?>
+<interfaceDefinition>
+  <node name="execute">
+    <properties>
+      <help>Initiate an operation</help>
```
Comment (dmbaturin) [inline op-mode-definitions/execute-port-scan.xml.in]: Basic node settings like this should only be in its top-level template, we shouldn't duplicate them.

File: op-mode-definitions/execute.xml.in [hunk 2]
```diff
@@ -1,8 +0,0 @@
-<?xml version="1.0"?>
```
Comment (dmbaturin) [inline op-mode-definitions/execute.xml.in]: Let's leave the file so that other `execute` commands can reuse it.

**Review by dmbaturin — CHANGES_REQUESTED** [review]

File: op-mode-definitions/execute-port-scan.xml.in [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    <children>
+      <node name="port-scan">
+        <properties>
+          <help>Scan network for open ports</help>
+        </properties>
+        <children>
+          <tagNode name="host">
+            <properties>
+              <help>IP address or domain name of the host to scan (scan all ports 1-65535)</help>
+			  <completionHelp>
```
Comment (sever-sever) [inline op-mode-definitions/execute-port-scan.xml.in]: the number of spaces does not match

**Review by sever-sever — CHANGES_REQUESTED** [review]

**Review by sever-sever — APPROVED** [review]

**Review by dmbaturin — APPROVED** [review]

**Comment by dmbaturin** [comment]
@mergifyio backport circinus

**Pull request merged**